### DISCUSSION TITLE
How to change the connection string based on the requested tenant?

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#6962] [Q&A] [answered]
I am attempting to use OrchardCore as a multi-tenant app in a non-CMS app. How can I change the connection string based on the requested tenant? I tried the following but it is not working "meaning the custom connection string isn't overriding the default"

```
services.AddDbContext<ApplicationContext>(options =>
{
    // Register the default database connection
    string connectionString = Configuration.GetConnectionString("DefaultConnection");
});

services.AddOrchardCore()
        .AddMvc()
        .WithTenants()
        .ConfigureServices((tenantServices, serviceProvider) =>
        {
            tenantServices.PostConfigure<DbContextOptionsBuilder>(options =>
            {
                // If the tenant has it's own instance of a database, we'll use it
                string connectionString = tenantConfigs.GetConnectionString("DefaultConnection");

                if (connectionString != null)
                {
                    // At this point we know that the tenant is shareing a database, we'll use the base connection
                    options.UseSqlServer(connectionString);
                }
            });

        });
```

Also, is there a service that I can use that would provide me with information about the current logged in tenant like some sort of identifier without using the `IShellConfiguration`, somrthing like `TenantAccessor`?

**Comment by deanmarcussen** [comment, marked as answer]
This is all very well documented https://docs.orchardcore.net/en/dev/docs/reference/core/Configuration/

You need to use `IShellConfiguration` if you want tenant level settings.

`.GetConnectionString` has no meaning in this context, you should load values from `IShellConfiguration` `appsettings` keys.

You can also inject `ShellSettings` to get settings for a specific tenant.

**Reply by ApacheTech** [reply]
That link is now dead.

**Reply by MikeAlhayek** [reply]
https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/
